### PR TITLE
Add basic criterion benches, update to 2018 version.

### PR DESCRIPTION
**Pull request by DevinR528**
Sorry about doing more than one thing in this PR it took me a while to get criterion happy. I realized I was trying to use 2018 edition imports. If the update is not wanted I will start a new PR with only criterion. I can either start working on benching and optimizing from here or start a separate PR.

### CODE REVIEW

File: src/keys.rs [hunk 1]
```diff
@@ -151,6 +151,25 @@ impl TrieKey for Path {
...
+            .encode_bytes()
+    }
+}
+
+#[cfg(windows)]
+impl TrieKey for Path {
+    fn encode_bytes(&self) -> Vec<u8> {
+        self.to_str().unwrap().as_bytes().encode_bytes()
+    }
+}
```
Comment (DevinR528) [inline src/keys.rs]: My thinking here is that if the `Trie` gets a `Path` or `PathBuf` at this point they will already be valid, `Path` and `PathBuf` would have failed if not valid? I kinda forgot I did this I wasn't sure if I should push and forgetting solved my problem cus I did it anyway.

Comment (DevinR528) [reply src/keys.rs]: While getting more familiar with the library I took a look at `NibbleVec` I added `criterion` to that too. Ill make a PR if you are interested. I was able to get some decent perf gains on everything but the `Into` and `From` traits. I used `SmallVec` of size 20, I can abstract this out with a trait so the user can choose size. Any thoughts to a good size for the `Trie`'s `NibbleVec`? I will try out my fork of [NibbleVec](https://github.com/DevinR528/rust_nibble_vec/tree/smallvec) in here and see if it makes a difference.

NIBBLE VEC PERF 
```bash
   Compiling nibble_vec v0.0.4 (/home/devinr/aprog/rust/__forks__/rust_nibble_vec)
    Finished bench [optimized] target(s) in 3.81s
     Running target/release/deps/nib_bench-d0739dc3a2efe765
nibvec split even       time:   [56.704 ns 56.752 ns 56.803 ns]                              
                        change: [-41.993% -41.902% -41.798%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 1 outliers among 100 measurements (1.00%)
  1 (1.00%) high mild

nibvec split odd        time:   [53.972 ns 54.067 ns 54.172 ns]                             
                        change: [-24.199% -24.055% -23.909%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 14 outliers among 100 measurements (14.00%)
  7 (7.00%) low severe
  2 (2.00%) low mild
  4 (4.00%) high mild
  1 (1.00%) high severe

join even nibvec to odd nib                                                                            
                        time:   [89.156 ns 89.511 ns 90.007 ns]
                        change: [-33.486% -32.970% -32.194%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 2 outliers among 100 measurements (2.00%)
  2 (2.00%) high severe

nib get on vec of 9 elements                                                                             
                        time:   [4.6243 ns 4.6254 ns 4.6267 ns]
                        change: [-68.638% -68.620% -68.600%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 9 outliers among 100 measurements (9.00%)
  3 (3.00%) low mild
  6 (6.00%) high mild

nib from vec and into vec                                                                             
                        time:   [45.542 ns 45.581 ns 45.621 ns]
                        change: [+29.644% +29.771% +29.895%] (p = 0.00 < 0.05)
                        Performance has regressed.
Found 4 outliers among 100 measurements (4.00%)
  1 (1.00%) low mild
  3 (3.00%) high mild

bench eq and not eq     time:   [272.67 ps 272.74 ps 272.80 ps]                                
                        change: [-97.862% -97.833% -97.806%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 10 outliers among 100 measurements (10.00%)
  3 (3.00%) low mild
  5 (5.00%) high mild
  2 (2.00%) high severe
```

Comment (DevinR528) [reply src/keys.rs]: A very rough no tinkering run adding `#[inline]` and switching out `NibbleVec` for `SmallVec<[u8; 64]> NibbleVec`
```bash
    Finished bench [optimized] target(s) in 4.01s
     Running target/release/deps/trie_benches-eea567fb604f9dca
trie insert             time:   [54.371 us 54.397 us 54.422 us]                        
                        change: [-0.9870% -0.6254% -0.4227%] (p = 0.00 < 0.05)
                        Change within noise threshold.
Found 1 outliers among 100 measurements (1.00%)
  1 (1.00%) high mild

trie get                time:   [23.483 us 23.490 us 23.497 us]                      
                        change: [-19.328% -19.284% -19.244%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 4 outliers among 100 measurements (4.00%)
  1 (1.00%) low severe
  2 (2.00%) low mild
  1 (1.00%) high severe

trie remove             time:   [83.044 us 83.067 us 83.089 us]                        
                        change: [-12.563% -12.531% -12.501%] (p = 0.00 < 0.05)
                        Performance has improved.
Found 4 outliers among 100 measurements (4.00%)
  3 (3.00%) low mild
  1 (1.00%) high severe
```

Comment (michaelsproul) [reply src/keys.rs]: I don't think the `Path`/`PathBuf` implementations are guaranteed not to panic, so I'd prefer if they were removed from this PR. Happy to try and work out a good implementation for them in a separate PR.

**Pull request merged**